src/main/java/ui/components/NavigableListView.java

```diff
@@ -216,4 +222,11 @@ public void selectNextItem() {
         return selectedIndex.map(getItems()::get);
     }
 
+    /**
+     * @return The integer value of the selected index in the list panel or -1 if it is null.
+     */
+    public int getSelectedIndex() {
+        return selectedIndex.isPresent() ? selectedIndex.get() : -1;
+    }
```

Comment:
You need to find out the index to know where to start iterating over the items that need to be marked read/unread.